db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -27,68 +31,105 @@ def request_totals(template_totals, dialog_totals)
+    request_hash_value(args_hash)
+  when :number_of_cpus
+    requested_number_of_cpus(args_hash)
+  when :storage
+    requested_storage(args_hash)
+  else
+    options_value(args_hash)
+  end
+  options_array
+end
+
+def vm_provision_cloud?
+  @miq_request.source.respond_to?(:cloud) ? @miq_request.source.cloud : false
+end
+
+def flavor_obj(id)
+  vmdb_object(:flavor, id)
+end
+
 def vm_prov_option_value(prov_option, options_array = [])
-  number_of_vms = get_option_value(@miq_request, :number_of_vms)
+  args_hash = {:prov_option   => prov_option,
+               :array         => options_array,
+               :resource      => @miq_request,
+               :flavor        => flavor_obj(@miq_request.get_option(:instance_type)),
+               :number_of_vms => get_option_value(@miq_request, :number_of_vms),
+               :cloud         => vm_provision_cloud?}
+
   case prov_option
   when :vm_memory
-    memory_in_request = number_of_vms * get_option_value(@miq_request, :vm_memory)
-    set_requested_value(prov_option, memory_in_request,
-                        @miq_request.get_option(:instance_type), options_array)
+    args_hash[:prov_value] = args_hash[:number_of_vms] * get_option_value(@miq_request, :vm_memory)
+    request_hash_value(args_hash)
   when :number_of_cpus
-    requested_number_of_cpus(number_of_vms, options_array)
+    requested_number_of_cpus(args_hash)
   when :storage
-    requested_storage(prov_option, number_of_vms, options_array)
+    requested_storage(args_hash)
   else
-    options_value(prov_option, @miq_request, options_array)
+    options_value(args_hash)
   end
   options_array
 end
 
-def requested_number_of_cpus(number_of_vms, options_array)
-  cpu_in_request = get_option_value(@miq_request, :number_of_cpus)
+def requested_number_of_cpus(args_hash)
+  cpu_in_request = get_option_value(args_hash[:resource], args_hash[:number_of_cpus])
   if cpu_in_request.zero?
-    cpu_in_request = get_option_value(@miq_request, :number_of_sockets) *
-                     get_option_value(@miq_request, :cores_per_socket)
+    cpu_in_request = get_option_value(args_hash[:resource], :number_of_sockets) *
+                     get_option_value(args_hash[:resource], :cores_per_socket)
   end
-  total_cpu = number_of_vms * cpu_in_request
-  set_requested_value(prov_option, total_cpu,
-                      @miq_request.get_option(:instance_type), options_array)
+  args_hash[:prov_value] = args_hash[:number_of_vms] * cpu_in_request
+  request_hash_value(args_hash)
 end
 
 def bytes_to_megabytes(bytes)
   bytes / 1024**2
 end
 
-def requested_storage(prov_option, number_of_vms, options_array)
-  vm_size = bytes_to_megabytes(@miq_request.vm_template.provisioned_storage)
-  total_storage = number_of_vms * vm_size
-  set_requested_value(prov_option, total_storage,
-                      @miq_request.get_option(:instance_type), options_array)
+def vmdb_object(object, id)
+  $evm.vmdb(object).find_by_id(id)
```

Comment:
Change this to be `$evm.vmdb(object, id)`.  Which is consistent with the other `$evm.vmdb` call in this method and I prefer that calling style for ID lookups.